src/ServiceControl/Monitoring/IMonitoringDataStore.cs

```diff
@@ -11,5 +11,6 @@ interface IMonitoringDataStore
         Task UpdateEndpointMonitoring(EndpointDetails endpoint, bool isMonitored);
         Task WarmupMonitoringFromPersistence(EndpointInstanceMonitoring endpointInstanceMonitoring);
         Task Delete(Guid endpointId);
+        Task<KnownEndpoint[]> GetAllKnownEndpoints();
```

Comment:
I think this is more aligned with other APIs:

```suggestion
        Task<IReadonlyList<KnownEndpoint>> GetAllKnownEndpoints();
```